File: src/test/java/com/ject/studytrip/stamp/helper/StampTestHelper.java
package com.ject.studytrip.stamp.helper;

import com.ject.studytrip.stamp.domain.model.Stamp;
import com.ject.studytrip.stamp.domain.repository.StampRepository;
import com.ject.studytrip.stamp.fixture.StampFixture;
import com.ject.studytrip.trip.domain.model.Trip;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

@Component
public class StampTestHelper {
    @Autowired private StampRepository stampRepository;

    public Stamp saveStamp(Trip trip, int order) {
        Stamp stamp = StampFixture.createStamp(trip, order);
        return stampRepository.save(stamp);
    }

    public Stamp saveDeletedStamp(Trip trip, int order) {
        Stamp stamp = StampFixture.createStamp(trip, order);
        stamp.updateDeletedAt();

        return stampRepository.save(stamp);
    }
}

File: src/test/kotlin/com/ject/studytrip/stamp/helper/StampTestHelper.kt
package com.ject.studytrip.stamp.helper

import com.ject.studytrip.stamp.domain.model.Stamp
import com.ject.studytrip.stamp.domain.repository.StampRepository
import com.ject.studytrip.stamp.fixture.StampFixture
import com.ject.studytrip.trip.domain.model.Trip
import org.springframework.stereotype.Component

@Component
class StampTestHelper(
    private val stampRepository: StampRepository,
) {
    fun saveStamp(
        trip: Trip,
        order: Int,
    ): Stamp = stampRepository.save(StampFixture(trip, order).create())

    fun saveDeletedStamp(
        trip: Trip,
        order: Int,
    ): Stamp = stampRepository.save(StampFixture(trip, order).create().also { it.updateDeletedAt() })

    fun saveCompletedStamp(
        trip: Trip,
        order: Int,
    ): Stamp = stampRepository.save(StampFixture(trip, order).create().also { it.updateCompleted() })

    fun getStamp(stampId: Long): Stamp = stampRepository.findById(stampId).orElseThrow()
}
